### PR TITLE
keyless authenticator fixes: b64 fixes, training wheels signature

### PR DESCRIPTION
**Pull request by alinush**
### Description

 - [x] JWT header in `KeylessSignature` should **not** be b64url-encoded
 - [x] Rename `exp_timestamp_secs` to `exp_date_secs` for consistency with our terminology
 - [x] JWT payload and signature in `OpenIdSig` should **not** be b64url-encoded
 - [x] Training wheels signature should cover ZK relation statement
 - [x] Add `_json` suffix
 
 No passkey support, because it would require working around the need for two passkey signatures: one one the TXN and another on the ZKP (non-malleability)

### CODE REVIEW

**Review by rex1fernando — APPROVED** [review]

File: types/src/keyless/openid_sig.rs [hunk 1]
```diff
@@ -113,15 +112,15 @@ impl OpenIdSig {
...
-        &self,
-        rsa_jwk: &RSA_JWK,
-        jwt_header_b64: &String,
-    ) -> anyhow::Result<()> {
-        let jwt_payload_b64 = &self.jwt_payload_b64;
-        let jwt_sig_b64 = &self.jwt_sig_b64;
-        let jwt_token = format!("{}.{}.{}", jwt_header_b64, jwt_payload_b64, jwt_sig_b64);
-        rsa_jwk.verify_signature(&jwt_token)?;
+    /// `jwt_header` is the *decoded* JWT header (i.e., *not* base64url-encoded)
+    pub fn verify_jwt_signature(&self, rsa_jwk: &RSA_JWK, jwt_header: &str) -> anyhow::Result<()> {
```
Comment (heliuchuan) [inline types/src/keyless/openid_sig.rs:116]: call jwt_header_json?

File: types/src/keyless/test_utils.rs [hunk 2]
```diff
@@ -3,11 +3,12 @@
 use crate::{
     jwks::rsa::RSA_JWK,
     keyless::{
-        base64url_encode_bytes, base64url_encode_str,
+        base64url_encode_str,
         circuit_testcases::{
             SAMPLE_EPK, SAMPLE_EPK_BLINDER, SAMPLE_ESK, SAMPLE_EXP_DATE, SAMPLE_EXP_HORIZON_SECS,
             SAMPLE_JWK, SAMPLE_JWK_SK, SAMPLE_JWT_EXTRA_FIELD, SAMPLE_JWT_HEADER_B64,
-            SAMPLE_JWT_PARSED, SAMPLE_PEPPER, SAMPLE_PK, SAMPLE_PROOF, SAMPLE_UID_KEY,
+            SAMPLE_JWT_HEADER_DECODED, SAMPLE_JWT_PARSED, SAMPLE_JWT_PAYLOAD_DECODED,
```
Comment (heliuchuan) [inline types/src/keyless/test_utils.rs:10]: I prefer _JSON to maintain the 

_B64 vs _JSON 

decoded lil ambiguous to me

for HEADER and PAYLOAD

**Review by perryjrandall — APPROVED** [review]

**Review by heliuchuan — APPROVED** [review]

File: types/src/keyless/openid_sig.rs (hunk 1 above)
Comment (alinush) [reply types/src/keyless/openid_sig.rs:116]: will do it later
need this finish CI/CD & merge

File: types/src/keyless/test_utils.rs (hunk 2 above)
Comment (alinush) [reply types/src/keyless/test_utils.rs:10]: Again, will fix this later, since we need to merge to release v1.10

Comment (alinush) [reply types/src/keyless/test_utils.rs:10]: Fixed now.

File: types/src/keyless/openid_sig.rs (hunk 1 above)
Comment (alinush) [reply types/src/keyless/openid_sig.rs:116]: Fixed now.

File: types/src/keyless/openid_sig.rs [hunk 3]
```diff
@@ -113,10 +114,10 @@ impl OpenIdSig {
     }
 
     /// `jwt_header` is the *decoded* JWT header (i.e., *not* base64url-encoded)
```
Comment (heliuchuan) [inline types/src/keyless/openid_sig.rs:116]: update the doc too

**Review by hariria — COMMENTED** [review]
LGTM overall. Seems like we may not do this since do signatures are required (one over the ZKP and another over the txn).

One thing I would do is just double check MAX_SIZE of the ZkidSignature to make sure it can accommodate a potentially 1024 byte Passkey signature inside.

File: types/src/keyless/openid_sig.rs (hunk 3 above)
Comment (alinush) [reply types/src/keyless/openid_sig.rs:116]: fudge, i'll have to do this later; don't wanna delay CI/CD (assuming this time it passes)

Comment (alinush) [reply types/src/keyless/openid_sig.rs]: fixed

**Pull request merged**